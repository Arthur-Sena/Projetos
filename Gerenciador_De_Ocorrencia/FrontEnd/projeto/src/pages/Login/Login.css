.App {
    max-height: 100vh;
    max-width: 130vh;
    overflow: hidden;
}

.AppPrincipal{
    max-height: 100vh;
    max-width: 130vh;
}

.Login {
    -webkit-transform: skew(35deg); 
      -ms-transform: skew(35deg);
          transform: skew(35deg); 
    /* z-index: 2; */
    background-color: white;
    margin-left: -30%;
    min-width: 64%;
}

.img{
    -webkit-transform: skew(-35deg); 
      -ms-transform: skew(-35deg); 
          transform: skew(-35deg); 
    margin-left: 16%;
    height: 20%;
    margin-top: 20px;
}

.DivLogin {
    /*------ inicio posicao ------ */
    background-color: white;
    position: relative;
    top: -550px;
    height: 100%;
    width: 67%;
    margin-left: 55%;
    margin-top: 5%;
    /* ------ Fim posicao ------ */
    border-color: black;
    border-style: solid;
    border-width: 1.5px;

}

.btnn {
    background-color: #005BA8;
    color: white;
    border-color: #005BA8;
    height: 15%;
    width: 25%;
    position: absolute;
    margin-left: 36%;
    font-size: x-large;
    margin-top: -20px;
}

.form {
    width: 80%;
}

.form input{
    width: 90%;
    margin-left: 15%;
    height: 35px;
    /* box-shadow: 0 0 0 0; */
    border-top: 0 none;
    border-left: 0 none;
    border-right: 0 none;
    /* border-bottom: 10px; */
    /* outline: 0; */
    border-color: grey;
    font-size: xx-large;
}