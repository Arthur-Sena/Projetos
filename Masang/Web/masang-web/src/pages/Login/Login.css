.App {
    position:absolute;
    background-image: url('../../assets/backgroundlogin1.png');
    background-size: 100vw 100vh;
    height: 100vh;
    width: 100vw;

}


.Login {
    Color: white;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
}


.h1Login {
    text-decoration: underline;
    text-decoration-color:  #f28123;
    font-weight: lighter;
}

.form {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
}

/* inputs */
.item {
    display: flex;
    flex-direction: column;
    border-style: none;
    background-color:rgba(f,f,f,f.2);
    border-bottom-color: white;
    border-bottom-style: ridge;
    width: 40vw;
}

.input__login{
    height:2em;
    border: none;
    background-color:rgba(0,0,0,0.01);
    color: white;
    font-size: large;
    outline:none;
    transition: border ease 2s;
}

.input__login:focus {
    border-bottom:solid 1px greenyellow
}

/* Botao */
.btn {
    height: 5vh;
    width: 9vw;
    background-color: #f28123;
    color: white;
    border-radius: 15px;
    font-size: medium;
    border-color: white;
    border-width: 1px;
    outline:none;
}

/* Css da Logo */
.Masang {
    /* margin-bottom: -1vh; */
    /* height:15vh; */
    margin-top: 13.5vh;
    margin-right: 2vw;
    display: flex;
    flex-direction: row;
    justify-content: flex-end;
    align-items: flex-end;
}

@media (max-width:1017px){
    .item {
        width: 70vw;
    }
    .btn {
        width: 20vw;
    }
}
@media (max-width:700px){
    .btn {
        width: 25vw;
    }

    .Masang {
        justify-content: center;
    }
}