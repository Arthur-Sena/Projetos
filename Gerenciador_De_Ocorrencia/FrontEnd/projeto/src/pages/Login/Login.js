import React, { Component } from 'react';

import './Login.css';
import Axios from 'axios';


import Back from '../../assets/img/login.JPG';
import senai from "../../assets/img/LogoSenai.jpg";


export default class Login extends Component {

    constructor() {
        super();
        this.state = {
            NIF: "",
            senha: "",
            erro: " ",
        }
    }

    //Inicio Salvar Senha e Nif no State
    loginNIF = (event) => {
        this.setState({ NIF: event.target.value })
    }

    loginSenha = (event) => {
        this.setState({ senha: event.target.value })
    }
    //Fim
    efetuarLogin = (event) => {
        event.preventDefault();

        Axios.post("http://192.168.5.158:5000/api/usuarios/login",{
                NIF: this.state.NIF,
                senha: this.state.senha
            })
            .then(data => {
                if (data.status === 200) {
                    localStorage.setItem("usuario-senai", data.data.token);

                    this.props.history.push('/Main');

                } else {
                    console.log("Erroooo")
                }
            })
            .catch(erro => {
                this.setState({ erro: "UsuÃ¡rio ou senha invÃ¡lidos" });
                console.log(erro);
            })
    }


    render() {

        return (
            <div className="AppPrincipal">

                <div className="App" style={{ display: 'flex' }}>
                    <img src={Back} style={{ marginTop: -167 }} />
                    <div className="Login" >
                        <img src={senai} className="img" />
                    </div>
                </div>
                {/* asssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssssss */}
                <div className="DivLogin"  >

                    <form onSubmit={this.efetuarLogin} className="form">
                        {/* <h1 style={{ textAlign: "center" }}>Login</h1> */}
                        <br></br>
                        <br></br>
                        <br></br>
                        <div className="inputLogin">
                            <input
                                placeholder="NIF"
                                type="text"
                                name="username"
                                onChange={this.loginNIF}
                                value={this.state.NIF}
                            />
                        </div>
                        <br></br>
                        <br></br>
                        <div className="inputLogin">
                            <input
                                placeholder="SENHA"
                                type="password"
                                name="password"
                                onChange={this.loginSenha}
                                value={this.state.senha}
                            />

                        </div>
                        <br></br>
                        <p
                            style={{ color: "red", marginLeft:'40%',fontSize:'large' }}
                        >
                            {this.state.erro}
                        </p>
                        <br></br>
                        <br></br>
                        <button className="btnn">
                            Login
                        </button >
                    </form>

                </div>
            </div>
        );
    }
}