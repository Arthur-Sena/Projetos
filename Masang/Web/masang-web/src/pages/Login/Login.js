import React, { Component } from 'react';
import Axios from 'axios';
import qs from 'qs';
//Css + images
import './Login.css';
import Logo from '../../assets/logoMasangofc1.png'

// const params = new URLSearchParams();
// params.append("username", this.state.email);
// params.append("password", password);
// params.append("scope", "api1");
// params.append("client_id", "ro.client");
// params.append("client_secret", secret);
// params.append("grant_type", "password");


class Login extends Component {

  constructor() {
    super();
    this.state = {
      email: "",
      senha: "",
      erro: ""
    }
  }

  loginEmail = (event) => {
    this.setState({ email: event.target.value })
  }

  loginSenha = (event) => {
    this.setState({ senha: event.target.value })
  }

  efetuarLogin = (event) => {
    event.preventDefault();

    if (this.state.email === 'admin@masang.com' && this.state.senha === '321321') {
      this.props.history.push('/services');
      localStorage.setItem("usuario-masang", 'admin');
    } else {
      this.setState({ erro: "Usu치rio ou senha inv치lidos" });
    }

    // const requestBody = {
    //     username: this.state.email,
    //     password: this.state.senha,
    //     scope: 'api1',
    //     client_id: 'ro.client',
    //     client_secret: 'secret',
    //     grant_type: 'password'
    // }

    // Axios.post("https://masang-id-dev.azurewebsites.net/connect/token", {
    //   body:  qs.stringify(requestBody),
    //   headers: {
    //     "Content-Type": "application/x-www-form-urlencoded",
    //   },
    // })
    //   .then(response => {
    //     if (response.status === 200) {
    //       // localStorage.setItem("usuario-masang", response.access_token);
    //       this.props.history.push('/services');
    //     } else {
    //       console.log(response)
    //     }
    //   })
    //   .then(data => {
    //     if (data.status === 200) {
    //       // localStorage.setItem("usuario-masang", data.body.access_token);
    //       this.props.history.push('/services');
    //     } else {
    //       console.log(data)
    //     }})
    //   .catch(erro => {
    //     this.setState({ erro: "Usu치rio ou senha inv치lidos" });
    //     console.log(erro);
    //   })
  }



  render() {
    return (
      <div className="App" >

        <div className="Login">

          <h1 className="h1Login" style={{ marginTop: '2em' }}>Login </h1>

          <form onSubmit={this.efetuarLogin} className="form">

            <div className="item" style={{ marginTop: '2em' }}>
              <label htmlFor="username" value="E-mail" style={{ fontSize: '17px' }}>E-mail</label>

              <input
                className="input__login"
                // placeholder="Email"
                type="email"
                name="username"
                onChange={this.loginEmail}
                value={this.state.email}
              />
            </div>

            <div className="item" style={{ marginTop: '4em' }}>
              <label htmlFor="password" value="label" style={{ fontSize: '17px' }}>Senha</label>

              <input
                className="input__login"
                // placeholder="Senha"
                type="password"
                name="password"
                onChange={this.loginSenha}
                value={this.state.senha}
              />
            </div>
            <div style={{ height: '11vh' }}>

              <p className="testJest" style={{ color: "white", textAlign: "center", fontSize: '20px', }}    >
                {this.state.erro}
              </p>

            </div>
            <button className="btn" type='submit' >
              Entrar
            </button >

          </form>
        </div>

        <div className="Masang">
          {/* <br></br> */}
          <img src={Logo} style={{ height: '15vh' }} />
        </div>

      </div>
    );
  }
}

export default Login;