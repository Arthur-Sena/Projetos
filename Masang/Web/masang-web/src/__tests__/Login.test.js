import React from 'react';
import renderer from 'react-test-renderer';
import Login from '../pages/Login/Login';
import { shallow, mount } from 'enzyme';
import '@testing-library/jest-dom/extend-expect'


//TESTES INSTÂNTANEOS
describe('Login Component [LOGINSCREEN]', () => {

    let component;

    beforeEach(() => {
        component = shallow(<Login />);
    });

    test('deve injetar corretamente um elemento', () => {
        const componenteInput = renderer.create(<input />).toJSON();
        expect(componenteInput).toMatchSnapshot();
    });

    test('deve executar a class e ciclos de vida sem crashar', () => {
        const classLogin = mount(<Login />);
        expect(classLogin).toMatchSnapshot();
    });

    test('deve executar o componente pai sem crashar', () => {
        expect(component).toMatchSnapshot();
    });

    test('deve iniciar váriáveis iniciais do state como vazio', () => {
        //console.log('valor state: ' + '"' + component.state("email") + '"' );
        expect(component.state('email')).toBe("");
        expect(component.state('senha')).toBe("");
        expect(component.state('erro')).toBe("");

    });

    test('deve renderizar ambos componentes inputs', () => {
        expect(component.find('input').length).toBe(2);

        expect(component.find('input[name="username"]').length).toBe(1);
        expect(component.find('input[name="password"]').length).toBe(1);
    });

    test('deve renderizar <p/> com mensagem de erro vazia', () => {
        expect(component.find('p[className="testJest"]').length).toBe(1);
        expect(component.find('p[className="testJest"]').text()).toBe('');
    });

    test('deve renderizar form  e button submit', () => {
        expect(component.find('form').length).toBe(1);
        expect(component.find('button').length).toBe(1);
    });

});

//TESTES DE COMPONENTES
describe('Inputs components [LOGINSCREEN]', () => {

    test('deve possuir dois labels não vazios', () => {
        expect(shallow(<Login />).find('label').length).toBe(2);

        expect(shallow(<Login />).find('label').at(0).text()).toBe('E-mail');
        expect(shallow(<Login />).find('label').at(1).text()).toBe('Senha');
    });

    describe('input type email', () => {
        let component;
        let inputEmail;

        //Executa a cada test do describe
        beforeEach(() => {
            component = shallow(<Login />);
            inputEmail = component.find('input').at(0);
        });

        test('deve conter props type e name pré setados', () => {
            expect(inputEmail.prop('type')).toBe('email');
            expect(inputEmail.prop('name')).toBe('username');
        });

        test('deve simular o onChange alterando o state', () => {
            expect(inputEmail.prop('value')).toBe('');

            const eventoFake = {
                target: {
                    value: 'testjest@gmail.com'
                }
            };

            //SIMULA ONCHANGE
            const onChangeNome = inputEmail.prop('onChange');
            onChangeNome(eventoFake);

            expect(component.state('email')).toBe('testjest@gmail.com');
            // expect(inputEmail.prop('value')).toBe('testjest@gmail.com');       
            // console.log(component.findWhere((node) => true).map(n => n.name()))
        });

    });

    describe('input type password', () => {
        let component;
        let inputPassword;

        beforeEach(() => {
            component = shallow(<Login />);
            inputPassword = component.find('input').at(1);
        });

        test('deve conter props type e name pré setados', () => {
            expect(inputPassword.prop('type')).toBe('password');
            expect(inputPassword.prop('name')).toBe('password');
        });

        test('deve simular o onChange alterando o state', () => {
            expect(inputPassword.prop('value')).toBe('');

            const eventoFake = {
                target: {
                    value: '123123'
                }
            };

            //SIMULA ONCHANGE
            const onChangeNome = inputPassword.prop('onChange');
            onChangeNome(eventoFake);

            expect(component.state('senha')).toBe('123123');
        });

    });

});

describe('Alterações forms [LOGINSCREEN]', () => {

    let component;
    let form;
    let history;

    beforeEach(() => {
        component = shallow(<Login history={history} />);
        form = component.find('form');
        history = { push: jest.fn() };
    });

    test('deve aparecer mensagem de erro por invalidez de login', () => {
        let errorMessage = component.find('p[className="testJest"]');

        const eventoFn = { preventDefault: () => { } }
        expect(errorMessage.text()).toBe("");

        jest.spyOn(eventoFn, 'preventDefault')
        form.simulate('submit', eventoFn);

        // console.log(component.state().erro)      
        expect(component.state().erro).toBe("Usuário ou senha inválidos");

    });

    //developing... Ao logar redirect p services (A FAZER QUANDO INTEGRAR REQUEST C UNIVERSUM)
    //redirecionar para ServicesScreen
    test('deve simular envio do formulário', () => {

        const eventoPrevDefault = { preventDefault: () => { } }
        const mockUser = {
            email: 'teste@teste.com',
            senha: '123'
        }

        component.setState({ email: mockUser.email, senha: mockUser.senha })
        expect(component.state('email')).toBeDefined();
        expect(component.state('email')).toBe(mockUser.email);
        expect(component.state('senha')).toBeDefined();
        expect(component.state('senha')).toBe(mockUser.senha);

        jest.spyOn(eventoPrevDefault, 'preventDefault')
        form.simulate('submit', eventoPrevDefault);

        //manipular prop. e ver se redireciona        
        // history.push('/Services');
        // expect(history.push).toHaveBeenLastCalledWith('/Services')
        
    })

});




