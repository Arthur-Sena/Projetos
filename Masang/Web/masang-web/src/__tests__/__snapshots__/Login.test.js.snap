// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component [LOGINSCREEN] deve executar a class e ciclos de vida sem crashar 1`] = `ReactWrapper {}`;

exports[`Login Component [LOGINSCREEN] deve executar o componente pai sem crashar 1`] = `ShallowWrapper {}`;

exports[`Login Component [LOGINSCREEN] deve injetar corretamente um elemento 1`] = `<input />`;
